src/telegraf.ts

```diff
@@ -69,34 +59,25 @@ namespace Telegraf {
   }
 }
 
-const allowedUpdates: tt.UpdateType[] | undefined = undefined
-
-export class Telegraf<
-  TContext extends Context = Context
-> extends Composer<TContext> {
-  private readonly options: Telegraf.Options<TContext>
+export class Telegraf<C extends Context = Context> extends Composer<C> {
+  private readonly options: Telegraf.Options<C>
   private webhookServer?: http.Server | https.Server
+  private polling?: Polling
   /** Set manually to avoid implicit `getMe` call in `launch` or `webhookCallback` */
   public botInfo?: tt.UserFromGetMe
   public telegram: Telegram
-  readonly context: Partial<TContext> = {}
-  private readonly polling = {
-    allowedUpdates,
-    limit: 100,
-    offset: 0,
-    started: false,
-    stopCallback: noop,
-    timeout: 30,
-  }
-
-  private handleError: (err: unknown, ctx: TContext) => void = (err: any) => {
-    console.error()
-    console.error((err.stack ?? err.toString()).replace(/^/gm, '  '))
-    console.error()
+  readonly context: Partial<C> = {}
+
+  private handleError = async (err: unknown, ctx: C): Promise<void> => {
+    // set exit code to emulate `warn-with-error-code` behavior of
+    // https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode
+    // to prevent a clean exit despite an error being thrown
+    process.exitCode = 1
+    debug('Middleware threw', err)
```

Comment:
Perhaps this is redundant now, when `handleError` doesn't do much, and `confirmUpdates` cannot replace the error?

Reply:
Yes.
```suggestion
```